e2e/utils/_.bash

```diff
@@ -34,6 +34,7 @@ standard_setup() {
 
     cd "$x/working-dir" || exit
 
+    export CARGO_HOME="$HOME/.cargo"
```

Comment:
Two environment variables that might be interesting in this context:
* `CARGO_NET_OFFLINE=true` will try to work offline, using cached data, and only access the network if it really needs to.
* `CARGO_NET_RETRY=10` will try harder if the network is inaccessible.  Sometimes CI fails for me due to a "spurious network error" and hitting retry passes, but this wastes a lot of time compared with just trying a bit harder to download rust packages.  Of course, if the local cache is used more, CI will download less so this will be an issue less frequently.

See: https://doc.rust-lang.org/cargo/reference/config.html#netretry